site-builder.js

```diff
@@ -116,7 +116,9 @@ SiteBuilder.prototype.build = function() {
 
 SiteBuilder.prototype.syncRepo = function() {
   this.logger.log('syncing repo:', this.repoName);
-  return this.spawn(this.git, ['pull']);
+  var that = this;
+  return this.spawn(this.git, ['pull'])
+    .then(function() { that.spawn(that.git, ['stash']); });
```

Comment:
Calling `that.spawn` directly will initiate the Promise right away, rather than waiting for the first Promise to resolve. Even though right now I got the tests to pass locally with that change, I remember that race issue biting me earlier.